frontend/src/main/scala/bloop/engine/Interpreter.scala

```diff
@@ -280,7 +285,23 @@ object Interpreter {
     }
   }
 
+  private def validateOutputPath(output: AbsolutePath, state: State, extension: String): Option[AbsolutePath] =
+    if (output.isDirectory) {
+      state.logger.error(s"The output path ${output.syntax} does not point to a file. Using default...")
+      None
+    } else if (!Files.isWritable(output.getParent.underlying)) {
+      state.logger.error(s"The output path ${output.getParent.syntax} cannot be created. Using default...")
+      None
+    } else if (!output.syntax.endsWith(extension)) {
+      // This is required for the Scala.js linker, otherwise it will throw an exception
+      state.logger.warn(s"The output path ${output.syntax} must have the extension '$extension'. Appending...")
+      Some(output.getParent.resolve(output.toFile.getName + extension))
+    } else Some(output)
```

Comment:
I would completely fail the execution if the validation fails, instead of writing the output somewhere the user doesn't expect it.

Reply:
You are right. The behaviour should be more deterministic. I will change it.